### PR TITLE
Modernize code to Python 3.6+

### PR DESCRIPTION
**Pull request by eumiro**
## Description

A series of independent commits to modernize the code to Python 3.6. Partially from `pyupgrade --py36-plus`, partially from PyCharm suggestions, partially manual find&replace. The individual commits contain very similar small diffs, so please process them commit after commit. If you prefer, I can split them into individual PRs.

1. `class A(object)` can be written as `class A:`
2. replace `dict([…])` and `set([…])` with `{…}`
3. use f-strings or compact `.format`
4. use `yield from` instead of `yield` in a `for` loop
5. import `mock` from `unittest`
6. expect `OSError` instead of `IOError` or `select` error
7. use Python3 defaults for file reading or `super()`
8. remove redundant parenthesis (keep those in tuples though)
9. shorten set intersection instead of creating lists
10. backslashes in strings do not have to be escaped if prepended with `r`

I would like to hear your opinion on different points and would be happy to adjust them to your needs.

## Checklist

- [ ] I've added this contribution to the `changelog.rst`.
- [ ] I've added my name to the `AUTHORS` file (or it's already there).

- [x] I installed pre-commit hooks (`pip install pre-commit && pre-commit install`), and ran `black` on my code.
- [x] Please squash merge this pull request (uncheck if you'd like us to merge as multiple commits)

### CODE REVIEW

**Comment by eumiro** [comment]
Rebased, repushed. Or would you prefer to split the commits to individual PRs?

File: tests/features/environment.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def before_all(context):
 def show_env_changes(env_old, env_new):
     """Print out all test-specific env values."""
     print("--- os.environ changed values: ---")
-    all_keys = set(list(env_old.keys()) + list(env_new.keys()))
+    all_keys = set(env_old) | set(env_new)
```
Comment (ruohola) [inline tests/features/environment.py:125]: Maybe cleaner to just use `env_old.keys() | env_new.keys()`?

**Comment by ruohola** [comment]
> Rebased, repushed.

Rebase once more and change the one commented line :) Seems to be really nice work! 

>  Or would you prefer to split the commits to individual PRs?

No need to split, I'll just squash merge this so I think it's perfect as is.

File: tests/features/environment.py (hunk 1 above)
Comment (eumiro) [reply tests/features/environment.py]: :+1:

**Comment by eumiro** [comment]
Thank you for the review, @ruohola. Finished, rebased, repushed.

**Comment by ruohola** [comment]
@eumiro Thanks for the contribution! :)

File: pgcli/packages/parseutils/tables.py [hunk 2]
```diff
@@ -42,8 +42,7 @@ def extract_from_part(parsed, stop_at_punctuation=True):
     for item in parsed.tokens:
         if tbl_prefix_seen:
             if is_subselect(item):
-                for x in extract_from_part(item, stop_at_punctuation):
-                    yield x
+                yield from extract_from_part(item, stop_at_punctuation)
```
Comment (amjith) [inline pgcli/packages/parseutils/tables.py:45]: This is a new one for me. :clap:

File: tests/features/steps/basic_commands.py [hunk 3]
```diff
@@ -97,9 +97,9 @@ def step_see_error_message(context):
 @when("we send source command")
 def step_send_source_command(context):
     context.tmpfile_sql_help = tempfile.NamedTemporaryFile(prefix="pgcli_")
-    context.tmpfile_sql_help.write(b"\?")
+    context.tmpfile_sql_help.write(br"\?")
     context.tmpfile_sql_help.flush()
-    context.cli.sendline("\i {0}".format(context.tmpfile_sql_help.name))
+    context.cli.sendline(fr"\i {context.tmpfile_sql_help.name}")
```
Comment (amjith) [inline tests/features/steps/basic_commands.py:102]: I didn't know we can combine the f-string and r-strings. :+1:

**Review by amjith — COMMENTED** [review]
Very nice!

**Pull request merged**